### ISSUE TITLE
Commented lines in host file are currently not ignored

### ISSUE THREAD

**Issue by Thomas499** [blokadaorg/blokada#372]
**Bug:** Commented out URL's in host files are not ignored. 

**Repeatable experiment that verifies this:** 
Step 1. Enable only https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts in host so that Steven Blacks list is the only host enabled. 
Step 2. Note how many host are currently blocked in blokada. For example, 37,471 are currently blocked. 
Step 3. Whitelist spclient.wg.spotify.com
Step 4. Note how many host are blocked in blokada now. For example, 37,470 (one less after spclient.wg.spotify.com was whitelisted)
Step 5. Go to https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts and verify if spclient.wg.spotify.com was commented out. 

**Expected Results:** Because spclient.wg.spotify.com was commented out of the host file, the URL should be ignored (not blocked by blokada). 
![image](https://user-images.githubusercontent.com/38442058/56298667-67ff3080-60e7-11e9-9606-d2f5526ed2a2.png)
Therefor, regardless if spclient.wg.spotify.com is whitelisted, the number of host blocked by blokada should be the same because # before the URL is supposed to tell blokada to ignore the following text as a comment. 

**Actual Results:** 
Adding spclient.wg.spotify.com to whitelist results in one less host being blocked by blokada. 

Issue should be marked as highest priority and warrants a new version of blokada being deployed **immediately** that fixes the documented bug for following reasons:

**Reason 1. Will break a LOT of apps** Host files often comment out URLs that are known to break apps. 
Blokada not ignoring these commented out URL's will result in the host files breaking each and every app that the URL's were commented out to not break by the host files. 

**Reason 2. Not fixing this bug will result in EVERY USER that has blokada with Steven Blacks host list enabled being banned from Spotify**
![image](https://user-images.githubusercontent.com/38442058/56299475-f3c58c80-60e8-11e9-9ea1-40e7a0832b1d.png)
If the commented out URL in the Steven Black list was ignored
![image](https://user-images.githubusercontent.com/38442058/56299516-0c35a700-60e9-11e9-962b-51f7028b3aa0.png)
as it should be ignored, then user would not be banned. But because blokada doesn't ignore the commented out URL, Spotify will ban the user. 

**Reason 3. If blokada doesn't ignore #'s it will run into issues as comments (text, messages in the host files).**

One such example of an issue would be if you use a custom URL in place of line 57 (filters %= URL) in  app/src/legacy/kotlin/core/Pages.kt and you replace it with a custom GIST file. 
To make it easy, enter the URL of this gist https://gist.githubusercontent.com/Thomas499/1117d60b8fa4dc2ba9bc59b4e3538e5d/raw into line 57 and you will notice that it works. 
as shown below
![image](https://user-images.githubusercontent.com/38442058/56302550-dbf10700-60ee-11e9-9cc6-a9ec74f1a41c.png)

Then enter the URL of this gist into line 57 https://gist.githubusercontent.com/Thomas499/214e682dc3c7e54d1dde9ad0bae8599f/raw and you will notice even though the only difference is a comment was added
![image](https://user-images.githubusercontent.com/38442058/56301274-5704ee00-60ec-11e9-9cbe-6f3822ba41b2.png)
It breaks the code so that no host show up in the blacklist by default. All you will get is a black screen as shown below. 
![image](https://user-images.githubusercontent.com/38442058/56302718-31c5af00-60ef-11e9-9021-dc9b5c9c8a42.png)

In this gist file, even though it isn't documented in the code that each default list needs to be exactly 7 lines of text followed by 2 blank lines, with the exception there needs to be 3 blank lines at the end, we simply experimented with comments and found that adding a comment to the code results in no default blacklist. So we have verified that Peter was correct. If Blokada doesn't ignore comments (starts with #) then it will run into issues if comments are used in the files. 
![image](https://user-images.githubusercontent.com/38442058/56299867-bad9e780-60e9-11e9-9f3e-155529718c8a.png)

**Reason 4. It is standard practice to ignore comments from hosts files.** Every other ad blocker ignores comments from host files because it is the standardized universally agreed good practice. 

Where code is believed to be BUT DOESN'T WORK AS IT SHOULD IN THEORY as the repeatable experiment listed above proves: 
https://github.com/blokadaorg/blokada/blob/master/app/src/legacy/kotlin/filter/HostlineProcessor.kt#L10
![image](https://user-images.githubusercontent.com/38442058/56303317-8584c800-60f0-11e9-962c-79f81683708f.png)

**Comment by Thomas499** [comment]
The problem we thought was not the actual problem. Discussion below:

![image](https://user-images.githubusercontent.com/38442058/56308573-76a41280-60fc-11e9-93a5-bf47a40b9114.png)

![image](https://user-images.githubusercontent.com/38442058/56309734-f9c66800-60fe-11e9-8013-250deae5fda5.png)
![image](https://user-images.githubusercontent.com/38442058/56309767-106cbf00-60ff-11e9-86c0-fbebc85b5c9a.png)

Problem is that if you whitelist a URL, even if the URL isn't blocked, it subtracts 1 from the number of host blokada says is blocked. I whitelisted Yourmom.com and verified that blokada subtracted one from total number of host it said it blocked. Interestingly, I then added Yourmom.com to a custom whitelist gist file that I activated and noticed that blokada is smart enough to not count the same whitelisted URL twice, but blokada isn't currently smart enough to first identify if the whitelisted URL was blocked before it subtracts 1 from the total blocked count. Starting a new issue for this subject. 

Note: issue mentioned in Reason 3 in first post still remains.

**Comment by rpeter85** [comment]
Just to have a clear view, does Blokada block the URL that is commented in the list file or the issue is that it is counted as 'blocked'?

**Comment by Thomas499** [comment]
> does Blokada block the URL that is commented in the list file
No

> or the issue is that it is counted as 'blocked'?
No

Issue was at first incorrectly interpreted as a much larger problem than it is. The only issue is documented thoroughly, as requested, in https://github.com/blokadaorg/blokada/issues/373 as a different issue now that we understand what is really happening.
